app/routes/home.js

```diff
@@ -8,25 +8,70 @@ import ENV from '../config/environment';
 let theme = {};
 
 export default Ember.Route.extend({
-    beforeModel(){
-        $.ajax({
-            type: "GET",
-            url: ENV.rootURL + "themes/theme_1.json",
-            async: false,
-            success: function (data) {
-                const content = Ember.Object.create(data);
-                const timeMachine = TimeMachine.Object.create({ content });
-                theme = timeMachine;
-            }});
+    setupController: function(controller, model) {
+        controller.set('model' , model);
     },
-    model(params){
+    model: async function(params){
+
         // If testing and parameter is not working use this 'jyu4t' for params.guid
-        return this.store.findRecord('node', params.guid).then((result)=>{
-            return {
-                theme,
-                guid: params.guid,
-                node: result
-            };
+        let node = await this.store.findRecord('node', params.guid)
+        let firebaseDB = this.store.findRecord('home', params.guid)
+
+        let jsonBlob = await firebaseDB.then((record)=>{        
+            return  record.get('jsonBlob')
+        },function(reason) {
+            return false;
         });
-    }
+
+
+
+
+        if(!jsonBlob){
+            if( node.get('currentUserPermissions')[1] === 'write'){
+                let defaultJSON ='';
+                $.ajax({
+                    type: "GET",
+                    url: ENV.rootURL + "themes/theme_1.json",
+                    async: false,
+                    success: function (data) {
+                        defaultJSON = data;
+                    }});
+
+
+                //add to firebase  
+                let guid = {
+                    id: params.guid,
+                    jsonBlob: JSON.stringify(defaultJSON) 
```

Comment:
This assignment is not waiting for the ajax call to finish. It looks like it should be moved to the success function or use the ajax promise.